File: picmaker.c
#include <stdio.h>
#include <fcntl.h>
#include <stdlib.h>
#include <unistd.h>

int main(){
  int fd = open("image.ppm", O_CREAT | O_TRUNC | 0666);
  
  int r, g, b, i;
  int width = 100;
  int height = 100;
  r = 255;
  g = 255;
  b = 255;
    
  char* s = "P3 400 400 255 ";

  write(fd,s, sizeof(s));

  for (i; i < 255; i-= 1){
    r = 255 -i; 
    write(fd, r ,sizeof(int));
    write(fd, g ,sizeof(int));
    write(fd, b ,sizeof(int));
    write(fd, " " ,sizeof(char));
  }
  // close(fd);
}

File: picmaker.py


def make_ppm():
    width = 400;
    height = 200;
    max_val = 255;
    
    r = 150;
    b = 255;
    g = 255;

  #  image = array.array('B', [r,g,b] *width * height)

    pic = open("pic.ppm", 'wb')
    pic.write("%s\n"% ("P3"))
    pic.write("%d %d\n"%(width,height))
    pic.write("%d\n"% (max_val))
    
    y = 0
    while(y < height):
        x = 0
        b-=1
        while (x < width):
            pic.write("%d %d %d " %(r,g,b));
            x+=1;

        y+=1;



make_ppm();
